Automattic/wp-calypso
# Design Picker: "Blog" category always appears below "Show All" in design picker sidebar

#### Changes proposed in this Pull Request

Categories are appearing in random order due to the fact that designs order is also randomised. This will allow us to see which themes are truly more popular and we can come up with an ordering after we get that info.
However we already know that bloggers make up a large percentage of users coming to wpcom, so it's already a safe bet to place that at the top (at least right below the "Show All" button). p1636700899343200-slack-C029SB8JT8S

This PR sorts categories so that "Blog" is always at the top.

I've also changed the way the `useCategorization()` hook worked. The logic for sorting belonged in the design picker step, since it was a flow specific thing. This meant pushing more control up the component tree, and required another identical prop that would need to be shared between both `useCategorySelection()` and `<DesignPicker>`. Passing 4 props that all worked together didn't feel right.

I took inspiration from [the way Reakit does it's state composition](https://reakit.io/docs/composition/), and changed `useCategorization()` to return a semi-opaque object that represents everything to do with categorisation. Then passing this object to `<DesignPicker>` both signals that categorisation should be enabled and provides all the necessary state so that the parent is in full control over how categorisation is done. The current Reakit API expects the state object to be "spread" when passed to the component, but Diego has rethought this: p9oQ9f-Iv-p2.

<img width="1817" alt="Screenshot 2021-11-17 at 5 46 10 PM" src="https://user-images.githubusercontent.com/1500769/142136458-d661d607-e2cb-466b-8690-8c061d47bcbc.png">


#### Testing instructions




* `calypso.localhost:3000/start/setup-site?siteSlug=<< testsite.wordpress.com >>`
* Test design picker in write and build flow
  * Blog should appear directly under Show All in both flows
  * Blog should be pre-selected in the write flow
  * Show All should be pre-selected the build flow
* Test DIFM flow
* Test the design picker in `/new`



Related to #
